mt-kahypar/datastructures/concurrent_bucket_map.h

```diff
@@ -69,8 +69,7 @@ class ConcurrentBucketMap {
  public:
 
   ConcurrentBucketMap() :
-    _num_buckets(align_to_next_power_of_two(
-      BUCKET_FACTOR * std::thread::hardware_concurrency())),
+    _num_buckets(align_to_next_power_of_two(BUCKET_FACTOR * BUCKET_FACTOR)),
```

Comment:
Is this change actually necessary? It seems to me that with the adjustment in `static_hypergraph` all code that uses this is already deterministic. I think it makes sense to scale the number of buckets with the number of threads

Reply:
I think this is necessary to get deterministic results across different machines. What do you think about letting the caller determine the value?

Reply:
Seems like a good idea. With regards to determinism, I thought that the algorithms that currently use the `ConcurrentBucketMap` are such that the output does not depend on the distribution to the buckets. If the caller determines the value, we can decide this for each algorithm separately :+1:

Reply:
I looked at this again and you were right. The two uses in StaticHypergraph::contract(..) don't depend on the distribution of elements to buckets. So I reverted the change :)